pkg/proc/gdbserial/gdbserver.go

```diff
@@ -1503,23 +1407,6 @@ func (t *Thread) reloadGAtPC() error {
 	cx := t.regs.CX()
 	pc := t.regs.PC()
 
-	// We are partially replicating the code of GdbserverThread.stepInstruction
```

Comment:
This code shouldn't be removed, removing it doesn't break debugserver on macOS because we can allocate memory on the target process but it will break lldb-server on linux and the spec is not clear on what would happen in this circumstance anyway.

Reply:
Is this due to lldb-server thinking that there is still a breakpoint (or breakpoints) present in this address range and changing the behavior of `step`?

Reply:
Looking into this more, is there really no way to get the value of `fs_base` out of LLDB/GDB? I know neither return the value in the registers command (CLI or wire protocol) but in GDB you can `print $fs_base`, so it has the information somewhere.

I'd like to get rid of having to special case this backend in terms of how we read into thread local storage to get the `*G`. I've done some digging in the documentation but wasn't able to find anything, however I didn't go too deep. I work with some GDB / LLDB maintainers so I think I might ping them on if there is a better way to get that value.

Reply:
> Looking into this more, is there really no way to get the value of fs_base out of LLDB/GDB?

GDB, yes it is actually in returned by the 'g' command (but this code doesn't work with gdb for other reasons). LLDB, no, as far as I know.

Reply:
PS. this would be a behaviour change so a refactoring wouldn't be the place to do it anyway.

Reply:
@aarzilli of course, I was planning on investigating this and then submitting as a separate patch.

Reply:
Coming back to this, I’m not sure I actually fully understand why this is an issue. Essentially we are clearing the breakpoints present in `[pc, pc+len(movinstr)]`, however we overwrite that memory anyways, meaning there will be no breakpoint instruction there. What’s the bug, exactly, that this prevents? Will lldb-server on linux complain and prevent us from writing to this memory if it knows there should be a breakpoint there?

Also, it seems lldb-server support on Linux is broken currently in master anyways (just ran `go test -v -backend=lldb` on my Fedora machine with failures) and we do not test it in Travis, so who knows for how long it's been broken. I will try and bisect to figure out what broke it, but regardless it is currently broken.

(On this note, I wish we could also test RR via CI, but Travis runs on GCE which does not support the counters RR needs).

Reply:
I actually just tried to bisect the lldb-server failures on linux, but I'm getting failures as far back as `v1.2.0`. 

Go v1.13
lldb-server v9.0.1

Reply:
I only have v6.0.0 installed, that one works. Regardless, this code has been here since the beginning, and the other branch depends on having memory allocation available, we don't know if that's available on every possible combination of lldb-server, debugserver and version of macOS so I wouldn't remove it.

In general, I'm not a fan of removing visibility into the list of breakpoints from the backends. It breaks this but this is just an example of things that we can't do if the backends can't se the breakpoint map. I like moving the breakpoint map into the target object but I'd also like to save a pointer to the target object inside gdbserial.Process and native.Process, so that we can access the state of Target from the backends when we need to.